ml-agents-envs/mlagents_envs/rpc_utils.py

```diff
@@ -43,24 +46,54 @@ def behavior_spec_from_proto(
 
 
 @timed
-def process_pixels(image_bytes: bytes, gray_scale: bool) -> np.ndarray:
+def process_pixels(image_bytes: bytes, expected_channels: int) -> np.ndarray:
     """
     Converts byte array observation image into numpy array, re-sizes it,
     and optionally converts it to grey scale
-    :param gray_scale: Whether to convert the image to grayscale.
     :param image_bytes: input byte array corresponding to image
+    :param expected_channels: Expected output channels
     :return: processed numpy array of observation from environment
     """
-    with hierarchical_timer("image_decompress"):
-        image_bytearray = bytearray(image_bytes)
-        image = Image.open(io.BytesIO(image_bytearray))
-        # Normally Image loads lazily, this forces it to do loading in the timer scope.
-        image.load()
-    s = np.array(image, dtype=np.float32) / 255.0
-    if gray_scale:
+    image_bytearray = bytearray(image_bytes)
+
+    if expected_channels == 1:
```

Comment:
@harperj Do you mind taking a look at this part? I made some changes to your hackweek code, because I was worried that `split()` might produce too many pieces in an edge case.

I'm still not sure the "right" way to do this; see https://stackoverflow.com/questions/63585016/determining-the-size-of-a-png-from-raw-bytes

Reply:
LGTM